src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -1801,14 +1801,17 @@ public ReceiveGossipType ReceiveGossip(GossipEnvelope envelope)
 
             if (remoteGossip.Equals(Gossip.Empty))
             {
-                _log.Debug("Cluster Node [{0}] - Ignoring received gossip from [{1}] to protect against overload",
+                _log.Debug("Cluster Node [{0}] - Ignoring empty received gossip from [{1}] to protect against overload",
                     _cluster.SelfAddress, from);
                 return ReceiveGossipType.Ignored;
             }
             if (!envelope.To.Equals(SelfUniqueAddress))
             {
-                _cluster.LogInfo("Ignoring received gossip intended for someone else, from [{0}] to [{1}]",
-                    from.Address, envelope.To);
+                _cluster.LogInfo("Ignoring received gossip intended for someone else, from [{0}] to [{1}]. Our full address is [{2}]",
+                    from.Address, envelope.To, SelfUniqueAddress);
+                
+                // TODO: if the gossip is received for a version of ourselves with a different UID, do we issue a Down command?
```

Comment:
Valid question - what we need to do here will be determined by what we see in the enhanced log messages, but I suspect in our cluster testing we're seeing some gossip for old incarnations of current nodes. Need to log the UID to be sure.